porcupine/tabs.py

```diff
@@ -413,6 +414,16 @@ class ReloadInfo(utils.EventDataclass):
 
 
 def _ask_encoding(path: pathlib.Path, encoding_that_didnt_work: str) -> str | None:
+    list_of_encodings = [x.replace("_", "-") for x in set(encodings.aliases.aliases.values())]
+
+    # must do this, because in list_of_encodings there are encodings, that codecs don't know
+    # but probably this isn't the appropriate place to do it
+    for encoding in list_of_encodings:
```

Comment:
This loop doesn't work. For a simpler example, consider:

```
>>> foo = ['a', 'b', 'c']
>>> for item in foo: foo.remove(item)
... 
>>> foo
['b']
```

The for loop internally creates a list iterator, which has an  index pointing into the list. Let's say that the index is `0`, and so we delete `foo[0]`, making `foo` to be `['b', 'c']`. So far so good. But then the index increments for the next iteration, and now we are deleting `foo[1]`, which is now `'c'` because `foo` is `['b', 'c']`. In general, loops like this can't delete two adjacent elements, because once an element is deleted, the index incrementing will cause the next element to be skipped.

To fix this, you can loop backwards, since deleting an element does not mess up indexes before whatever it is you deleted:

```suggestion
    for encoding in reversed(list_of_encodings):
```

Or you can loop over a copy of the list:

```suggestion
    for encoding in list_of_encodings.copy():
```

Reply:
Looking into this more, it seems like the right solution would be to use a hard-coded list: https://stackoverflow.com/a/25584253

Currently Python 3.7 is the oldest Python that Porcupine supports. If you add a comment that mentions `3.7`, the encoding list will get updated when Porcupine drops support for Python 3.7.